source/extensions/filters/common/ext_authz/ext_authz_impl.h

```diff
@@ -19,7 +18,12 @@
 
 #include "common/singleton/const_singleton.h"
 
+#include "extensions/filters/common/ext_authz/ext_authz.h"
+
 namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
```

Comment:
This works well for ext_authz, +1. I'm wondering how you think this structure plays out when we have two HTTP filters with common code (rather than a network and an HTTP file sharing some common code)? Do we still use `filters/common`, or is there `filters/http/common`?

Reply:
I was planning on doing `filters/http/common` and I would do `Extensions::HttpFilters::Common::`.

Reply:
I would find it really helpful if we had it documented somewhere what goes where and why, as part of this series of PRs.   Internally we have a requirements on BUILD files they have a top level comment on what goes where but many of our intermediate directories don't have BUILD files (yet).  I'm open to BUILD files in the "empty" directories, or FOO.md files, or a REPO_ORGANIZATION file in a top level directory, or comments on each module why they live where they live, whatever works!

Reply:
I'm going to go ahead and add a REPO_ORGANIZATION.md to the top level in this PR. I think that's a great idea.